src/app/shared/users.service.ts

```diff
@@ -1,33 +1,58 @@
 import { Injectable } from '@angular/core';
-import { usersList } from './users.mock';
 import { User } from './users.model';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
 export class UsersService {
 
-  constructor() { }
+  mainURL: string = 'http://localhost:8080/';
+  usersMainURL: string = `${this.mainURL}users/`;
 
-  getAllUsers(): User[] {
-    return usersList;
+  constructor(private http: HttpClient) { }
+
+  getAllUsers(): Observable<User[]> {
+    return this.http.get<User[]>(this.usersMainURL);
+  }
+
+  getActiveUsers(): Observable<User[]> {
+    return this.http.get<User[]>(this.usersMainURL).pipe(
+      map(usersList => usersList.filter(user => !user.isDeleted))
+    );
   }
 
-  getActiveUsers(): User[] {
-    return usersList.filter(user => user.isDeleted === false);
+  getDeletedUsers(): Observable<User[]> {
+    return this.http.get<User[]>(this.usersMainURL).pipe(
+      map(usersList => usersList.filter(user => user.isDeleted))
+    );
   }
 
-  getDeletedUsers(): User[] {
-    return usersList.filter(user => user.isDeleted === true);
+  getUserDetails(userId: string): Observable<User> {
+    return this.http.get<User>(this.usersMainURL + userId);
```

Comment:
You can use `${this.userMainURL}${userId}`

Applicable at all the other places as well

Reply:
did the changes